File: src/types/alert.ts
// Admin 告警设置 API 的 DTO 类型。
// 通知记录只含安全元数据；前端不构造、不展示任何内部标识（后端本就不返回）。

export interface AlertRuleDTO {
  id: string;
  rule_name: string;
  severity: string;
  threshold: number | null;
  threshold_unit: string | null;
  enabled: boolean;
  notification_channels: string[];
  dedup_strategy: string | null;
  updated_at: string;
}

export interface AlertRulesResponseDTO {
  items: AlertRuleDTO[];
}

export interface AlertRuleUpdateDTO {
  enabled?: boolean;
  threshold?: number;
  notification_channels?: string[];
}

export interface NotificationDTO {
  id: string;
  alert_rule_id: string | null;
  audit_event_id: string | null;
  recipient_user_id: string;
  recipient_name: string | null;
  channel: string;
  title: string;
  content: string;
  send_status: string;
  sent_at: string | null;
  created_at: string;
}

export interface NotificationsResponseDTO {
  items: NotificationDTO[];
}

File: backend/app/schemas/alert.py
"""Admin 告警设置 API 的请求 / 响应 schema。

alert_rules 承载归档阈值等可配置规则；notification_records 为本地通知记录，
只回安全元数据（标题 / 安全摘要内容），不回业务原文与内部敏感标识。
"""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel


class AlertRuleOut(BaseModel):
    id: uuid.UUID
    rule_name: str
    severity: str
    threshold: float | None
    threshold_unit: str | None
    enabled: bool
    notification_channels: list[str]
    dedup_strategy: str | None
    updated_at: datetime


class AlertRulesResponse(BaseModel):
    items: list[AlertRuleOut]


class AlertRuleUpdateBody(BaseModel):
    enabled: bool | None = None
    threshold: float | None = None
    notification_channels: list[str] | None = None


class NotificationOut(BaseModel):
    id: uuid.UUID
    alert_rule_id: uuid.UUID | None
    audit_event_id: uuid.UUID | None
    recipient_user_id: uuid.UUID
    recipient_name: str | None
    channel: str
    title: str
    content: str
    send_status: str
    sent_at: datetime | None
    created_at: datetime


class NotificationsResponse(BaseModel):
    items: list[NotificationOut]
